File: solutions/csharp/jedliks-toys/1/ElonsToys.cs
using System;

class RemoteControlCar
{
    private int _metersDriven;
    private int _batteryPercentage = 100;

    public static RemoteControlCar Buy() => new();

    public string DistanceDisplay() => $"Driven {_metersDriven} meters";

    public string BatteryDisplay()
    {
        if (_batteryPercentage == 0)
            return "Battery empty";
        return $"Battery at {_batteryPercentage}%";
    }

    public void Drive()
    {
        if (_batteryPercentage == 0) return;

        _metersDriven += 20;
        _batteryPercentage -= 1;
    }
}

File: solutions/go/jedliks-toys/1/elons_toys.go
// Package elon provides methods for cars.
package elon

import "fmt"

// Drive reduces the car's battery by the battery drain amount and increases the
// distance based on the car's speed.
func (car *Car) Drive() {
	if car.battery-car.batteryDrain < 0 {
		return
	}

	car.battery -= car.batteryDrain
	car.distance += car.speed
}

// DisplayDistance returns a string containing the current distance driven by the car.
func (car Car) DisplayDistance() string {
	return fmt.Sprintf("Driven %d meters", car.distance)
}

// DisplayBattery returns as tring containing the current battery percentage of the car.
func (car Car) DisplayBattery() string {
	return fmt.Sprintf("Battery at %d%%", car.battery)
}

// CanFinish determines if the car can finish the track without running out of battery.
func (car Car) CanFinish(trackDistance int) bool {
	return car.battery/car.batteryDrain*car.speed >= trackDistance
}
